torchvision/models/detection/fcos.py

```diff
@@ -501,7 +502,10 @@ def postprocess_detections(
                 topk_idxs = torch.where(keep_idxs)[0]
 
                 # keep only topk scoring predictions
-                num_topk = min(self.topk_candidates, topk_idxs.size(0))
+                if torchvision._is_tracing():
+                    _, num_topk = det_utils._onnx_tracing_topk_min(topk_idxs, self.topk_candidates, 0)
```

Comment:
Does this provide the wrong answer for non-tracing mode?
If not, why do we need the if-statement?

Reply:
I don't see why this change shouldn't work for both modes. 
@prabhat00155 Could you point me to the appropriate tests for detection models in tracing and scripting mode to test the above suggestion?

Reply:
Hi @shubhambhokare1 , Seems that torchvion now only uses tracing mode when exporting ONNX, I guess the following tests might be suitable for reference

https://github.com/pytorch/vision/blob/97eddc5d6a83a9bf620070075ef1e1864c9a68ac/test/test_onnx.py#L430-L444
https://github.com/pytorch/vision/blob/97eddc5d6a83a9bf620070075ef1e1864c9a68ac/test/test_onnx.py#L372-L386

Reply:
Seems like the call
```
cast(int, min_kval)
```
leads to the error
```
fn = torch._C._jit_script_compile(
RuntimeError: 
__int__ cannot be used as a value:
```
This might be a torchscript limitation, reverting back to different branches of scripting and tracing for now